### PR TITLE
#164069271 Users should be presented with a reset password link

### PR DESCRIPTION
**Pull request by NLSanyu**
**What does this PR do?**

* Allows users to reset their passwords

**Description of the tasks to be completed?**

* Add a link on the log in form that allows a user to reset their password

**How should this be manually tested?**
`(NOTE: You must be already registered with Author’s Haven in order to reset your password)`
* Navigate to the landing page on https://ah-front-end-groot.herokuapp.com/
* Click on the `Log in` button on the top right of the page.
* Click on the `Forgot password` link. This will navigate to the reset password page.
* Enter your email address and click `Send reset password link`. An email will be sent to you with a link to reset your password. Click on that link to be sent to the reset password page. 
* Enter your new password and click `Reset password` to reset your password.

image

```Provide email for reset password```
image

```Change the password```
image

**What are the relevant Pivotal Tracker stores?**
[#164069271](https://www.pivotaltracker.com/story/show/164069271)

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (Stanley-Okwii) [inline .vscode/launch.json]: @NLSanyu , Please do not commit this file.

File: src/actions/loginAction.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import axios from 'axios';
+
+import { LOGIN_SUCCESS, LOGIN_FAIL } from './types';
+import endPoints from '../containers/urls';
+import 'react-toastify/dist/ReactToastify.css';
```
Comment (Stanley-Okwii) [inline src/actions/loginAction.js]: I see two different toasts imported, please remove what you are not using

File: tests/reset_password/resetPasswordReducers.spec.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      let state = resetPasswordReducer(undefined, {});
+      expect(state).toEqual({});
+    });
+
+    it('should update state with reset password email', () => {
+      expect(resetPasswordReducer([], {type: types.PROVIDE_RESET_EMAIL, payload:'lnsanyu@yahoo.com'})).toEqual({
+        email: 'lnsanyu@yahoo.com'
+      })
+    });
+});
```
Comment (Stanley-Okwii) [inline tests/reset_password/resetPasswordReducers.spec.js]: Add a new line

**Review by Stanley-Okwii — APPROVED** [review]

File: src/actions/loginAction.js (hunk 2 above)
Comment (NLSanyu) [reply src/actions/loginAction.js]: This is from the Login feature that I rebased from. I believe it is better that @walimike decides how to deal with this.

File: tests/reset_password/resetPasswordReducers.spec.js (hunk 3 above)
Comment (NLSanyu) [reply tests/reset_password/resetPasswordReducers.spec.js]: Done

File: .vscode/launch.json (hunk 1 above)
Comment (NLSanyu) [reply .vscode/launch.json]: I've removed it

**Review by akram256 — APPROVED** [review]
Nice work @NLSanyu. Keep it up

**Review by walimike — APPROVED** [review]

**Comment by NLSanyu** [comment]
> Nice work @NLSanyu. Keep it up

Thanks @akram256

**Comment by nyabongoedgar** [comment]
LGTM

**Review by nyabongoedgar — APPROVED** [review]

**Review by huxaiphaer — APPROVED** [review]
LGTM

**Pull request merged**